libtransact/src/contract/context/error.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::error::Error as StdError;
+
+use crate::contract::address::AddresserError;
+use crate::protos::ProtoConversionError;
+
+#[derive(Debug)]
+pub enum KeyValueContextError {
+    AddresserError(AddresserError),
+    ProtoConversionError(ProtoConversionError),
+    ProtocolBuildError(Box<dyn StdError>),
+}
+
+impl StdError for KeyValueContextError {
+    fn source(&self) -> Option<&(dyn StdError + 'static)> {
+        match *self {
+            KeyValueContextError::AddresserError(_) => None,
+            KeyValueContextError::ProtoConversionError(ref err) => Some(err),
+            KeyValueContextError::ProtocolBuildError(ref err) => Some(&**err),
+        }
+    }
+}
+
+impl std::fmt::Display for KeyValueContextError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match *self {
+            KeyValueContextError::AddresserError(ref err) => write!(f, "Addresser Error: {}", err),
```

Comment:
Just like the previous commit, these errors should be human-readable rather than just restating the error variant.